xonsh/dirstack.py

```diff
@@ -0,0 +1,216 @@
+import os
+import builtins
+from argparse import ArgumentParser
+
+DIRSTACK = []
+
+
+def pushd(args, stdin=None):
+    global DIRSTACK
+
+    try:
+        args = pushd_parser.parse_args(args)
+    except SystemExit:
```

Comment:
Yes it does, actually!  If there is an error when parsing, `ArgumentParser` prints the help message and then exits using `SystemExit`.